lnwallet/wallet.go

```diff
@@ -881,6 +888,41 @@ func (l *LightningWallet) handleFundingReserveRequest(req *InitFundingReserveMsg
 			}
 		}
 
+		// Before we can provision a channel with provided outpoints we
```

Comment:
Won't this just fail later when we try to fetch the coin for the outpoint if it's unspent?

Reply:
I am not following. The `WithCoinSelectLock` releases the mutex after the passed in function ends. So the coins are available for selection later again. Is that what you mean?

Reply:
Ah now I see. Yes then I think the way to go is to check this within the same coin select within `ProvisionChannel` if the selected outpoints are still unspent.